x-pack/plugins/fleet/common/services/agent_status.ts

```diff
@@ -41,7 +41,7 @@ export function getAgentStatus(agent: Agent | FleetServerAgent): AgentStatus {
       ? agent.policy_revision_idx
       : undefined;
 
-  if (!policyRevision || (agent.upgrade_started_at && !agent.upgraded_at)) {
+  if (!policyRevision || (agent.upgrade_started_at && agent.upgrade_status !== 'completed')) {
```

Comment:
could it be simplified to `agent.upgrade_status === 'started'` ?

Reply:
one advantage of having a check for `not completed`, is that there might be new statuses later (e.g. `retrying` pretty soon) that would still mean the agent is `upgrading` (unless we want to change the status logic)